config/common.php

```diff
@@ -38,16 +39,18 @@
         ],
     ],
 
-    TransportInterface::class => $params['yiisoft/mailer']['useSendmail'] ? SendmailTransport::class : [
-        'class' => EsmtpTransport::class,
-        '__construct()' => [
-            $params['symfony/mailer']['esmtpTransport']['host'],
-            $params['symfony/mailer']['esmtpTransport']['port'],
-            $params['symfony/mailer']['esmtpTransport']['tls'],
-        ],
-        'setUsername()' => [$params['symfony/mailer']['esmtpTransport']['username']],
-        'setPassword()' => [$params['symfony/mailer']['esmtpTransport']['password']],
-    ],
+    TransportInterface::class => $params['yiisoft/mailer']['useSendmail']
+        ? SendmailTransport::class
+        : static function () use ($params): TransportInterface {
```

Comment:
In this case `EsmtpTransport` always will be without event dispatcher and logger.
[Here ](https://github.com/symfony/mailer/blob/c1f83da2296741110be35dd779f2a9e412cec466/Transport/Smtp/EsmtpTransportFactory.php#L30) `$this->dispatcher` and `$this->logger` always `null`.